File: Bridge/Kaliuzhnyi-03/ConsoleApp1/Shapes/Rectangle.cs
using Bridge.Colors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge.Shapes;
public class Rectangle : Shape
{
    protected readonly double _sideOne;
    protected readonly double _sideTwo;

    public Rectangle(double sideOne, double sideTwo, IColor color) : base(color)
    {
        if (sideOne <= 0 || sideTwo <= 0)
        {
            throw new ArgumentException("Error! Rectangle's side can't be smaller or equal to zero!");
        }

        _sideOne = sideOne;
        _sideTwo = sideTwo;
    }

    public override double Area()
    {
        return _sideOne * _sideTwo;
    }

    public override double Perimeter()
    {
        return (_sideOne + _sideTwo) * 2;
    }

    public override string ToString()
    {
        string rectangleToString = $"There is rectangle with sides: {_sideOne}, {_sideTwo}; color: {Color}";

        return rectangleToString;
    }

}

File: Bridge/Petukhov/src/lnu/edu/ua/shapes/Rectangle.java
package lnu.edu.ua.shapes;

import lnu.edu.ua.colors.Color;

public class Rectangle extends Shape {
    protected double firstSide;
    protected double secondSide;

    public Rectangle(Color color, double firstSide, double secondSide) {
        super(color);
        this.firstSide = firstSide;
        this.secondSide = secondSide;
    }

    public Double getPerimeter() {
        return firstSide*2 + secondSide*2;
    }

    public Double getArea() {
        return firstSide * secondSide;
    }
    public String toString() {
        StringBuilder str = new StringBuilder("Rectangle:\n");
        str.append("firstSide: ").append(this.firstSide).append("\n");
        str.append("secondSide: ").append(this.secondSide).append("\n");
        str.append("perimeter: ").append(getPerimeter()).append("\n");
        str.append("area: ").append(getArea()).append("\n");
        str.append("color: ").append(this.color.getColor()).append("\n");
        return str.toString();
    }
}
